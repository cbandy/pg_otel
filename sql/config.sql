-- vim: set expandtab shiftwidth=0 syntax=pgsql tabstop=2 :

-- TEST: custom parameters and their defaults
\pset format csv
SELECT name, setting, unit, context, vartype, min_val, max_val, enumvals
  FROM pg_catalog.pg_settings
 WHERE name LIKE 'otel.%';
\pset format aligned

-- TEST: endpoint requires scheme
ALTER SYSTEM SET otel.otlp_endpoint TO 'localhost:8080';

-- TEST: protocol cannot be changed
ALTER SYSTEM SET otel.otlp_protocol TO 'grpc';

-- TEST: service.name cannot be blank
ALTER SYSTEM SET otel.service_name TO '';
