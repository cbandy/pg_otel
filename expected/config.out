-- vim: set expandtab shiftwidth=0 syntax=pgsql tabstop=2 :
-- TEST: custom parameters and their defaults
\pset format csv
SELECT name, setting, unit, context, vartype, min_val, max_val, enumvals
  FROM pg_catalog.pg_settings
 WHERE name LIKE 'otel.%';
name,setting,unit,context,vartype,min_val,max_val,enumvals
otel.attribute_count_limit,128,,internal,integer,128,128,
otel.otlp_endpoint,http://localhost:4318,,sighup,string,,,
otel.otlp_protocol,http/protobuf,,internal,string,,,
otel.otlp_timeout,10000,ms,sighup,integer,1,3600000,
otel.resource_attributes,,,sighup,string,,,
otel.service_name,postgresql,,sighup,string,,,
\pset format aligned
-- TEST: protocol cannot be changed
ALTER SYSTEM SET otel.otlp_protocol TO 'grpc';
ERROR:  parameter "otel.otlp_protocol" cannot be changed
-- TEST: service.name cannot be blank
ALTER SYSTEM SET otel.service_name TO '';
ERROR:  invalid value for parameter "otel.service_name": ""
DETAIL:  resource attribute "service.name" cannot be blank.
